### PR TITLE
SW-974-fix-legacy image

### PR DESCRIPTION
**Pull request by JFSolution**
fix that the migration run for the legacy image
and fix beamos_version refactor bug

### CODE REVIEW

**Comment by ahmed-mrbeam** [comment]
I also discovered one more thing. Please see my comment here 
https://github.com/mrbeam/MrBeamPlugin/pull/1429/files

**Comment by JFSolution** [comment]
> I also discovered one more thing. Please see my comment here https://github.com/mrbeam/MrBeamPlugin/pull/1429/files

I can't see your comment

File: tests/migrations/test-migration-Mig001.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class TestMigrationMig001(unittest.TestCase):
+    """
+    Testclass for the migration Mig001
+    """
+    def setUp(self):
+        self.m001 = Mig001NetconnectdDisableLogDebugLevel(None)
+
+    def test_beamos_versions(self):
```
Comment (ahmed-mrbeam) [inline tests/migrations/test-migration-Mig001.py:12]: for the sake of completion, we should add an assert for non matching version string

File: octoprint_mrbeam/migration/Mig001.py [hunk 2]
```diff
@@ -14,6 +14,7 @@ class Mig001NetconnectdDisableLogDebugLevel(MigrationBaseClass):
     def __init__(self, plugin):
         """
         initalization of the migration 001
+
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/migration/Mig001.py:17]: would be nice if the  white space formatting here could be reverted

Comment (JFSolution) [reply octoprint_mrbeam/migration/Mig001.py:17]: this is needed to show the doc strings correctly, thats why I added it

File: octoprint_mrbeam/migration/migration_base.py [hunk 3]
```diff
@@ -73,13 +75,19 @@ def id(self):
...
         Checks if this Miration should run
+
         Args:
             cls: Migrationclass
             beamos_version: current beamos_version
 
         Returns:
             bool: True if this migration should run
         """
+        if not isinstance(beamos_version, basestring):
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/migration/migration_base.py:86]: Very nice selection of the class to compare to .. a link to confirm your good choice  
https://stackoverflow.com/a/1979107

**Review by ahmed-mrbeam — APPROVED** [review]

**Pull request merged**